controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/kvtable/DeleteTableTask.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor.requesthandlers.kvtable;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.controller.retryable.RetryableException;
+import io.pravega.controller.store.kvtable.KVTableMetadataStore;
+import io.pravega.controller.store.kvtable.KVTOperationContext;
+import io.pravega.controller.store.kvtable.KeyValueTable;
+import io.pravega.controller.store.stream.StoreException;
+import io.pravega.controller.task.KeyValueTable.TableMetadataTasks;
+import io.pravega.controller.util.RetryHelper;
+import io.pravega.shared.controller.event.kvtable.DeleteTableEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Request handler for performing scale operations received from requeststream.
+ */
+@Slf4j
+public class DeleteTableTask implements TableTask<DeleteTableEvent> {
+
+    private final KVTableMetadataStore kvtMetadataStore;
+    private final TableMetadataTasks kvtMetadataTasks;
+    private final ScheduledExecutorService executor;
+
+    public DeleteTableTask(final KVTableMetadataStore kvtMetaStore,
+                           final TableMetadataTasks kvtMetaTasks,
+                           final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(kvtMetaStore);
+        Preconditions.checkNotNull(kvtMetaTasks);
+        Preconditions.checkNotNull(executor);
+        this.kvtMetadataStore = kvtMetaStore;
+        this.kvtMetadataTasks = kvtMetaTasks;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteTableEvent request) {
+        String scope = request.getScope();
+        String kvt = request.getKvtName();
+        long requestId = request.getRequestId();
+        String kvTableId = request.getTableId().toString();
+
+        return RetryHelper.withRetriesAsync(() -> getKeyValueTable(scope, kvt)
+                .thenCompose(table -> table.getId()).thenCompose(id -> {
+            if (!id.equals(kvTableId)) {
+                return CompletableFuture.completedFuture(null);
```

Comment:
please log that this workflow completed because of this condition.

Reply:
fixed